src/Dapr.PluggableComponents/Adaptors/StateStoreAdaptor.cs

```diff
@@ -0,0 +1,247 @@
+/// </remarks>
+public class StateStoreAdaptor : StateStoreBase
+{
+    private readonly ILogger<StateStoreAdaptor> logger;
+    private readonly IDaprPluggableComponentProvider<IStateStore> componentProvider;
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="StateStoreAdaptor"/> class.
+    /// </summary>
+    /// <param name="logger">A logger used for internal purposes.</param>
+    /// <param name="componentProvider">A means to obtain the Dapr Pluggable Component associated with this adapter instance.</param>
+    /// <exception cref="ArgumentNullException"></exception>
+    public StateStoreAdaptor(ILogger<StateStoreAdaptor> logger, IDaprPluggableComponentProvider<IStateStore> componentProvider)
+    {
+        this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        this.componentProvider = componentProvider ?? throw new ArgumentNullException(nameof(componentProvider));
+    }
+
+    /// <inheritdoc/>
+    public override async Task<BulkDeleteResponse> BulkDelete(BulkDeleteRequest request, ServerCallContext context)
+    {
+        this.logger.LogInformation("BulkDelete request for {count} keys", request.Items.Count);
+
+        var stateStore = this.GetStateStore(context);
+
+        if (stateStore is IBulkStateStore bulkStateStore)
+        {
+            await bulkStateStore.BulkDeleteAsync(
+                request.Items.Select(StateStoreDeleteRequest.FromDeleteRequest).ToArray(),
+                context.CancellationToken);
+        }
+        else
+        {
+            this.logger.LogInformation("Store does not support bulk operations; falling back to individual operations.");
+
+            foreach (var item in request.Items)
+            {
+                await stateStore.DeleteAsync(StateStoreDeleteRequest.FromDeleteRequest(item), context.CancellationToken);
+            }
+        }
+
+        return new BulkDeleteResponse();
+    }
+
+    /// <inheritdoc/>
+    public override async Task<BulkGetResponse> BulkGet(BulkGetRequest request, ServerCallContext context)
+    {
+        this.logger.LogInformation("Bulk get request for {count} keys", request.Items.Count);
+
+        var stateStore = this.GetStateStore(context);
+
+        if (stateStore is IBulkStateStore bulkStateStore)
+        {
+            var items = await bulkStateStore.BulkGetAsync(
+                request.Items.Select(StateStoreGetRequest.FromGetRequest).ToArray(),
+                context.CancellationToken);
+
+            var response = new BulkGetResponse
+            {
+                Got = items.Any(item => String.IsNullOrEmpty(item.Error))
+            };
+
+            response.Items.Add(items.Select(StateStoreBulkStateItem.ToBulkStateItem));
+
+            return response;
+        }
+        else
+        {
+            this.logger.LogInformation("Store does not support bulk operations; falling back to individual operations.");
```

Comment:
Nit: This might be better as a debug as it could get pretty verbose if people are doing a lot of gets.

Reply:
Agree, changed all adaptor logging to "debug" level.